### ISSUE TITLE
SystemError: new style getargs format but argument is not a tuple

### ISSUE THREAD

**Issue by cristobal-ortega** [RRZE-HPC/pylikwid#13]
The test in the README fails with the following message:

```
SystemError: new style getargs format but argument is not a tuple
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
SystemError: <built-in function init> returned a result with an error set
```

Using Python 3.6.5.

When trying to initialize pylikwid with a different object, it just seg faults:
```
>>> import pylikwid
>>> pylikwid.init("0")
Segmentation fault
```

```
>>> tup = tuple([0,1,2])
>>> print (tup)
(0, 1, 2)
>>> import pylikwid
>>> pylikwid.init(tup)
Segmentation fault
```

Not sure, what is causing this. Any dependencies needed? (aside from likwid)

**Comment by TomTheBear** [comment]
Thanks for pointing this out. I'm not sure about the message and asked a colleague who does more Python stuff.

I think the second code snippet does not work due the unpacking of the list at the C-backend. I look into this.

**Comment by sv2518** [comment]
Hello. I was wondering how this has been resolved? I would like to use pylikwid, but I am having the exact same issue. I am using Python 3.6.9.

**Comment by TomTheBear** [comment]
I thought it was fixed in the commit https://github.com/RRZE-HPC/pylikwid/commit/91ba172a1b48fea40ca36d017745ec7194c5ea80 but I'll check again. 

Have you tried the master branch from the repo or installation via pip? If pip, which version?

**Comment by TomTheBear** [comment]
I checked the code. The `pylikwid.init()` call accepts a **list** of CPU IDs. I cannot find any example that uses tuples for the cpulist. But indeed, the code segfaults when providing the cpulist as tuple. I will add a check but if you use it properly, it should work.

Tested with Python 3.8.5

**Comment by sv2518** [comment]
> I thought it was fixed in the commit [91ba172](https://github.com/RRZE-HPC/pylikwid/commit/91ba172a1b48fea40ca36d017745ec7194c5ea80) but I'll check again.
> 
> Have you tried the master branch from the repo or installation via pip? If pip, which version?

I used pip install and then double checked that bit of code in the commit 91ba172 and it did. I had used pip 21.0.1. 

> I checked the code. The pylikwid.init() call accepts a list of CPU IDs. I cannot find any example that uses tuples for the cpulist. But indeed, the code segfaults when providing the cpulist as tuple. I will add a check but if you use it properly, it should work.

I was running the test ./tests/testlib.py, so I assume I was using the function properly.

So I took a step back and installed everything new, meaning I did a fresh install of likwid and then rather then using pip install, I installed pylikwid via the master branch from the repo. Now it works.

If you are not able to reproduce this, it was probably just a quirk in my system, so I am happy to close the issue. Thanks for your help.

**Comment by TomTheBear** [comment]
Thanks for reporting back.

I did some changes yesterday and check now the type of the given cpulist in `pylikwid.init()`:
```
>>> import pylikwid
>>> pylikwid.init((0,1,2))
No or wrong function argument: List required
TypeError: argument 1 must be list, not tuple
```
I worked generally on the argument handling between Python and C. The compiler warning should be gone now as well.
